libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Common/SFSDKWebViewStateManager.m

```diff
@@ -45,6 +45,7 @@ + (void)resetSessionWithNewAccessToken:(NSString *)accessToken isSecureProtocol:
 + (void)removeSession {
     //reset UIWebView related state if any
     [self removeUIWebViewCookies:@[SID_COOKIE] fromDomains:self.domains];
+    [self removeWKWebViewCookies:self.domains withCompletion:NULL];
```

Comment:
I'd like to see logout become asynchronous in the next major SDK release.  There are a number of internal cleanup processes to the SDK (like this one) that should be async, in addition to consumer-driven cleanup that would benefit from the same.  Logout can be a pretty significant event in the app lifecycle.  Apps would benefit from not having to try to serialize all of that state change before resetting.